agent/agent_test.go

```diff
@@ -670,9 +675,18 @@ func verifyIndexChurn(t *testing.T, tags []string) {
 		ServiceName: "redis",
 	}
 	var before structs.IndexedCheckServiceNodes
+
+	// This sleep is so that the serfHealth check is added to the agent
+	// A value of 375ms is sufficient enough time to ensure the serfHealth
+	// check is added to an agent. 500ms so that we don't see flakiness ever.
+	time.Sleep(500 * time.Millisecond)
+
```

Comment:
Sad but OK if this is the only thing that works here.

One question though - I see lots of different tests flaking out on `serfHealth` issue same as this. For example several did in my last PR test run that were not this test: https://travis-ci.org/hashicorp/consul/jobs/414102708

Even if sleep is the best we've got, is there a way to do it somewhere more central so that we don't have to copy paste this line and comment into lots of different tests?

Maybe we could try just adding it before the return o `NewTestAgent()` ? What do you think?

Reply:
Yeah, I'm game for it. Let me check it locally and report back if it works. That's ingenious if that manages to work.

Reply:
So I did the change, and 
```--- FAIL: TestAPI_SemaphoreForceInvalidate (1.52s)
        semaphore_test.go:85: err: failed to create session: Unexpected response code: 500 (Missing check 'serfHealth' registration)```
This was the first error I encountered. I don't think, the sleep fixes other tests. 😞